File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/d/ems/tmpl/ErrorEmsDTO.java
package iih.ci.ord.d.ems.tmpl;

import java.util.List;

import iih.ci.ord.d.ems.base.BaseCiDTO;
import iih.ci.ord.ems.d.CiEnContextDTO;
import xap.mw.core.data.FArrayList;

/**
 * 错误医疗单对象
 * @author wangqingzhu
 *
 */
public class ErrorEmsDTO extends BaseCiDTO {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	public ErrorEmsDTO(){}
	
	public ErrorEmsDTO(CiEnContextDTO ctx,Object docObj,List<String> msgList){
		setEnContext(ctx);
		FArrayList doc = new FArrayList();
		doc.add(docObj);
		setDocument(doc);
		FArrayList msg = new FArrayList();
		msg.addAll(msgList);
		setErrorInfo(msg);
	}

	/**
	 * 错误信息
	 * @return FArrayList
	 */
	public FArrayList getErrorInfo() {
		return ((FArrayList) getAttrVal("ErrorInfo"));
	}
	/**
	 * 错误信息
	 * @param ErrorInfo
	 */
	public void setErrorInfo(FArrayList ErrorInfo) {
		setAttrVal("ErrorInfo", ErrorInfo);
	}
	
	
}

File: client/iih_stub/iih.ci_stub/iih.ci.ord/dto/emsmain/tmpl/ErrorEmsDTO.cs
using iih.ci.ord.dto.emsmain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using xap.mw.core.data;

namespace iih.ci.ord.d.ems.tmpl
{
    public class ErrorEmsDTO : BaseCiDTO
    {
        /**
	 * 错误信息
	 * @return FArrayList
	 */
        public FArrayList ErrorInfo
        {
            get { return getAttrVal<FArrayList>("ErrorInfo", null); }
            set { setAttrVal<FArrayList>("ErrorInfo", value); }
        }
        
    }
}
